src/UCTNode.h

```diff
@@ -83,8 +83,8 @@ class UCTNode {
 
     // Tree data
     std::atomic<bool> m_has_children{false};
-    UCTNode* m_firstchild{nullptr};
-    UCTNode* m_nextsibling{nullptr};
+    std::vector<UCTNode*> m_children;
```

Comment:
So, one reason to use a vector is to get rid of the pointers. Unfortunately, due to the use of atomics, we can't move the nodes into the container (boo!). So we have to use pointers. But we ARE able to use std::unique_ptr inside the vector.

And using them would solve the gigantic memory leak that I think you have right now. You're not freeing the nodes, are you?

And actually, if we can avoid having to sort the nodes at any point, then you could actually store the nodes directly in the vectors. You removed one of those sorts already. Maybe it's possible to get rid of the second one too.

Reply:
I got rid of all references but emplace_back still require the move constructor :(